### PR TITLE
Allow pre preprocessing hooks to substitute for ops, adds onnx  global pooling averaging and max

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Allow pre preprocessing hooks to substitute for ops, adds global pooling averaging and max
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

## Quick checklist

The following checklist helps ensure your PR is complete:

- [ X] Eclipse Contributor Agreement signed, and signed commits - see [IP Requirements](https://deeplearning4j.org/eclipse-contributors) page for details
- [ X] Reviewed the [Contributing Guidelines](https://github.com/eclipse/deeplearning4j/blob/master/CONTRIBUTING.md) and followed the steps within.
- [ X] Created tests for any significant new code additions.
- [ X] Relevant tests for your changes are passing.

### CODE REVIEW

**Comment by agibsonccc** [comment]
@treo this one is also self contained, but depends on the changes here: https://github.com/eclipse/deeplearning4j/pull/9410

File: nd4j/samediff-import/pom.xml [hunk 1]
```diff
@@ -254,7 +254,7 @@
         </dependency>
         <dependency>
           <groupId>org.nd4j</groupId>
-          <artifactId>nd4j-native</artifactId>
+          <artifactId>nd4j-native-platform</artifactId>
```
Comment (treo) [inline nd4j/samediff-import/pom.xml]: We don't usually need to build and run the tests on multiple platforms from the same artifact, why add the platform dependency here?

File: nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/definitions/OnnxOpDeclarations.kt [hunk 2]
```diff
@@ -360,8 +360,18 @@ val gemm = OnnxMappingProcess(
...
+        inputFrameworkOpName = "GlobalAveragePool",
+        opMappingRegistry = onnxOpRegistry
+)
 //TODO: GlobalLpPool
 //TODO: GlobalMaxPool
+val globalMaxPooling = OnnxMappingProcess(
+        opName = "noop",
+        inputFrameworkOpName = "GlobalMaxPool",
+        opMappingRegistry = onnxOpRegistry
+)
```
Comment (treo) [inline nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/definitions/OnnxOpDeclarations.kt]: Is this done? If yes, remove the TODO markers, if not, add a note what is missing.

**Review by treo — CHANGES_REQUESTED** [review]

File: nd4j/samediff-import/pom.xml (hunk 1 above)
Comment (agibsonccc) [reply nd4j/samediff-import/pom.xml]: Removed. I think this was testing left overs.

File: nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/definitions/OnnxOpDeclarations.kt (hunk 2 above)
Comment (agibsonccc) [reply nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/definitions/OnnxOpDeclarations.kt]: Updated and clarified context around op.

**Review by treo — APPROVED** [review]

**Pull request merged**